Assets/HoloToolkit/SpatialSound/Scripts/AudioLoFiEffect.cs

```diff
@@ -0,0 +1,222 @@
+            if (fs != filterSettings)
+            {
+                // If we have an attached AudioInfluencerManager, we need to let it know
+                // about our filter settings change.
+                AudioInfluencerManager influencerManager = gameObject.GetComponent<AudioInfluencerManager>();
+                if (influencerManager != null)
+                {
+                    influencerManager.SetNativeLowPassCutoffFrequency(fs.LowPassCutoff);
+                    influencerManager.SetNativeHighPassCutoffFrequency(fs.HighPassCutoff);
+                }
+
+                filterSettings = fs;
+                lowPassFilter.cutoffFrequency = filterSettings.LowPassCutoff;
+                highPassFilter.cutoffFrequency = filterSettings.HighPassCutoff;
+            }
+        }
+
+        // Populates the source quality filter settings collection.
+        private void LoadQualityFilterSettings()
+        {
+            if (sourceQualityFilterSettings.Keys.Count > 0) { return; }
+
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.FullRange,
+                new FilterSettings(10, 22000));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.NarrowBandTelephone,
+                new FilterSettings(300, 3400));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.WideBandTelephone,
+                new FilterSettings(50, 7000));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.AmRadio,
+                new FilterSettings(40, 5000));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.FmRadio,
+                new FilterSettings(30, 15000));
+        }
+
+        /// <summary>
+        /// Source quality options that match common telephony and
+        /// radio broadcast options.
+        /// </summary>
+        public enum AudioSourceQuality
+        {
+            /// <summary>
+            /// Narrow frequency range telephony
+            /// </summary>
+            NarrowBandTelephone,
+
+            /// <summary>
+            /// Wide frequency range telephony
+            /// </summary>
+            WideBandTelephone,
+
+            /// <summary>
+            /// AM radio
+            /// </summary>
+            AmRadio,
+
+            /// <summary>
+            /// FM radio
+            /// </summary>
+            FmRadio,
+
+            /// <summary>
+            /// Full range of human hearing.
+            /// </summary>
+            /// <remarks>
+            /// The frequency range used is a bit wider than that of human
+            /// hearing. It closely resembles the range used for audio CDs.
+            /// </remarks>
+            FullRange
+        }
+
+        /// <summary>
+        /// Settings for the filters used to simulate a low fidelity sound source.
+        /// </summary>
+        public struct FilterSettings
```

Comment:
Please extract structs in to their own file, should never embed structures.

Reply:
I'll extract the enum and the struct from the class. They are specific to working with this class, so I'll leave them in the source file, but make their names clearly imply that they are for use with the AudioLoFiEffect.